pkg/pillar/base/execwrapper.go

```diff
@@ -134,6 +97,19 @@ func (c *Command) execCommand() ([]byte, error) {
 	}
 }
 
+// WithCustomTimeout set custom timeout for command
+func (c *Command) WithCustomTimeout(timeout time.Duration) *Command {
+	// $ grep hv_watchdog_timer pkg/grub/rootfs.cfg
+	const watchdogTimeout = 500 * time.Second
+
+	c.timeout = timeout
+	if c.timeout > watchdogTimeout {
+		c.errorMonad = fmt.Errorf("custom timeout (%v) is longer than watchdog timeout (%v)", c.timeout, watchdogTimeout)
```

Comment:
I'm just curious: what's wrong with the command execution timeout being longer than the watchdog timeout? Theoretically, a command can run much longer than a standard watchdog timeout (some storage-related workloads like encryption, conversion, or whatever): it runs in a dedicated thread, and the watchdog files are still being touched by the main service...

Reply:
It is okay if it is still able to trigger the watchdog, but if not then it is a problem.
Some commands run in a different goroutine; I run in this issue with `dhcpcd -U...`.

Reply:
see also https://github.com/lf-edge/eve/pull/3428/commits/046a629b40b20d5a50002c8a108e42d6b37924be

Reply:
@christoph-zededa do we need that commit as part of this PR to not (temporarily) decreasing the timeout for those potentially long operations?